### PR TITLE
implement feture 9967 : rados pool rollback

### PR DESCRIPTION
**Pull request by xinxinsh**
Signed-off-by: xinxin shu <xinxin.shu@intel.com>

### CODE REVIEW

**Comment by liewegas** [comment]
can you make a workunit that tests this?  create a pool, write some objects, snap, write more and overwrite, pool-rollback, and verify things are as they should be.  e.g., make sure objects created after the snap are deleted.

**Comment by xinxinsh** [comment]
@liewegas ,  i am not familiar with workunit tests , can you tell me where can i add these tests ,  adding this test into qa/workunits/rados/test_rados_tool.sh is ok to you ?

**Comment by liewegas** [comment]
Yeah, or Perhaps qa/workunits/rados/pool_rollback.sh or something like 
that.  It's a simple script that assumes a working/running environment, so 
you'll want it to create a pool, write some stuff and do the test, then 
remove the pool when it's done.  Include set -e in there so that it will 
exit with an error code if anything goes wrong.

**Comment by xinxinsh** [comment]
@liewegas , updated , pls help review

**Comment by xinxinsh** [comment]
@liewegas , since object list CLI can only list head objects, maybe this patch depends on #3213

**Comment by athanatos** [comment]
I don't think this is a good idea:
1) I think users would expect this to be atomic, which it is not.
2) It's intractable for any reasonably sized pool.
3) The same functionality can be created trivially using the python rados bindings.

File: src/tools/rados/rados.cc [hunk 1]
```diff
@@ -1555,6 +1557,27 @@ static int rados_tool_common(const std::map < std::string, std::string > &opts,
...
+      goto out;
+    }
+    if (nargs.size() < 2)
+      usage_exit();
+
+    librados::NObjectIterator i = io_ctx.nobjects_begin();
+    librados::NObjectIterator i_end = io_ctx.nobjects_end();
+    for (; i != i_end; ++i) {  
+      string oid = i->get_oid();
+      ret = io_ctx.snap_rollback(oid, nargs[1]);
```
Comment (dzafman) [inline src/tools/rados/rados.cc]: This would need to set the namespace before calling snap_rollback().

**Comment by athanatos** [comment]
I am closing this since it is intractable for any reasonably sized pool and would leave such a pool in an indeterminate state if canceled part way through.

**Pull request closed without merging**